atc/api/resourceserver/list_shared_for_resource.go

```diff
@@ -0,0 +1,45 @@
+package resourceserver
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/concourse/concourse/atc/db"
+	"github.com/tedsuo/rata"
+)
+
+func (s *Server) ListSharedForResource(pipeline db.Pipeline) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		resourceName := rata.Param(r, "resource_name")
+		logger := s.logger.Session("list-shared-for-resource", lager.Data{"resource-name": resourceName})
+
+		resource, found, err := pipeline.Resource(resourceName)
+		if err != nil {
+			logger.Error("failed-to-get-resource", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		if !found {
+			logger.Info("resource-not-found")
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		shared, err := resource.SharedResourcesAndTypes()
+		if err != nil {
+			logger.Error("failed-to-get-shared-resource-and-types", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		err = json.NewEncoder(w).Encode(shared)
+		if err != nil {
+			logger.Error("failed-to-encode-shared-resources-and-types", err)
+			w.WriteHeader(http.StatusInternalServerError)
```

Comment:
somewhere it returns error in the response body so fly cmd will show the error together with status code, which I think might be helpful. Have you considered to do the same for these new endpoints?

https://github.com/concourse/concourse/blob/c0e0782bfac8f345820c9a1631631fa03129e322/atc/api/resourceserver/check.go#L64

Reply:
Thats a good idea, let me add that!

Reply:
@clarafu are you going to add the error mesage to the response still?